tests/integration/test_extreme_deduplication/test.py

```diff
@@ -49,20 +49,25 @@ def test_deduplication_window_in_seconds(started_cluster):
     node.query("INSERT INTO simple VALUES (0, 1)")
     assert TSV(node.query("SELECT count() FROM simple")) == TSV("2\n")
 
-    # wait clean thread
-    time.sleep(5)
+    # Wait for the cleanup thread.
+    for i in range(100):
```

Comment:
There is a better way - `assert_eq_with_retry`

Or I added in the https://github.com/ClickHouse/ClickHouse/pull/52083 general `retry` module.